arch/arm/core/cortex_m/mpu/arm_mpu_v7_internal.h

```diff
@@ -106,6 +106,16 @@ static inline void _get_mpu_ram_region_attr(arm_mpu_region_attr_t *p_attr,
 	p_attr->rasr = _get_region_attr(1, ap, 1, 1, 1, 1, 0, size);
 }
 
+#ifdef CONFIG_RAM_FUNCTION
+static inline void _get_mpu_ramfunc_region_attr(arm_mpu_region_attr_t *p_attr,
+						u32_t base, u32_t size)
+{
+	ARG_UNUSED(base);
+
+	p_attr->rasr = _get_region_attr(0, RO, 0, 1, 0, 0, 0, size);
```

Comment:
I don't think this is correct. You indeed want to disable the eXecute Never bit, but you want to keep TEX=1, C=1 and B=1.

Reply:
You should consult the ARMv7-M architecture/technical reference manual.